symbolic-debuginfo/src/dwarf.rs

```diff
@@ -1348,10 +1356,35 @@ impl<'data> DwarfDebugSession<'data> {
     }
 
     /// See [DebugSession::source_by_path] for more information.
+    ///
+    /// Note that this does not load additional sources from disk and only works with sources
+    /// embedded directly in the debug information (DW_LNCT_LLVM_source)
     pub fn source_by_path(
         &self,
-        _path: &str,
+        path: &str,
     ) -> Result<Option<SourceFileDescriptor<'_>>, DwarfError> {
+        // First: see if there's an exact match for the provided path
```

Comment:
I believe its okay to use a `OnceLock` here to collect these into a `HashMap` for faster lookups.
Usually we want to optimize for lookups here.

Reply:
Okay :+1: 
I'd like to avoid collecting all source code in a big map of owned strings but I'm not sure if that's doable given that the `DwarfFileIterator<'_>` returned by `files(&self)` give out lifetimes shorter than `'data` IIUC.

Reply:
What's your take on this implementation? https://github.com/getsentry/symbolic/pull/849/files#diff-ffd2feab131b47be099bceda4b175052d29c014642276be0168bd3d782a569f9L1351

I'm using a map now, but still iterate over the DWARF units on lookup to avoid dealing with owned strings and lifetime issues.